### ISSUE TITLE
C_Cpp.intelliSenseCacheSize initially empty, causing language server crashes

### ISSUE THREAD

**Issue by npnkhoi** [microsoft/vscode-cpptools#11426]
### Environment

- OS and Version: Ubuntu 22.04.2
- VS Code Version: 1.82.0
- C/C++ Extension Version: 1.17.5
- If using SSH remote, specify OS of remote machine: N/A

### Bug Summary and Steps to Reproduce

Bug Summary:
When I download this extention (`ms-vscode.cpptools`) and open a simple .C file, the language server says it crashes for 5 times in 3 minutes. After that, nothing about this extention works.

Looking at the settings of the extension, I saw that `C_Cpp.intelliSenseCacheSize` was empty. However, this field is required (VsCode show a red warning). When I fill the value `5120` to it and reload VsCode, the error disappears and everything works normally.

Steps to reproduce:
1. On a new VsCode install on an environment similar to mine, download and install `ms-vscode.cpptools`.
2. Open a simple `.C` file (e.g., hello world)
4. See error -- the language server crashes.

Expected behavior:
The language server should not crash by default like that.

### Configuration and Logs

```shell
{
  "configurations": [
    {
      "name": "linux-gcc-x64",
      "includePath": [
        "${workspaceFolder}/**"
      ],
      "compilerPath": "/usr/bin/gcc",
      "cStandard": "${default}",
      "cppStandard": "${default}",
      "intelliSenseMode": "linux-gcc-x64",
      "compilerArgs": [
        ""
      ]
    }
  ],
  "version": 4
}

// I cannot retrieve the Output now because I cannot clear the value of C_Cpp.intelliSenseCacheSize after setting it to a number.
```

**Comment by sean-mcmanus** [comment]
What exactly do you mean by "empty"?

**Comment by npnkhoi** [comment]
This is what I meant by "empty". I saw it like this when I was going through the settings of the extension. After I set it to 5120, the crashes went away.

![image](https://github.com/microsoft/vscode-cpptools/assets/11204020/d3653706-159f-4d48-ac8a-9113b25bb091)

**Comment by sean-mcmanus** [comment]
@npnkhoi Can you click the Open JSON settings button in the top right of the settings UI? That will show what is actually being used.

**Comment by sean-mcmanus** [comment]
@npnkhoi I just reproed the crash via setting it to a string instead of a number.

**Comment by sean-mcmanus** [comment]
@npnkhoi But if your setting is set to something other than a string, let us know so we can handle that case too.

**Comment by sean-mcmanus** [comment]
@npnkhoi FYI, this general issue is known to potentially be problematic/crash-causing and is tracked by https://github.com/microsoft/vscode-cpptools/issues/11375 .

**Comment by sean-mcmanus** [comment]
@npnkhoi Also, I assume you or something else must have set that to a string? I don't know how it could default to an invalid value. Maybe the workspace you opened had that setting already set?

**Comment by npnkhoi** [comment]
Yes, it is possible that I might set it to some illegal values in JSON while messing around with the settings some time ago, but I cannot be sure because once I set it back to a valid number via GUI, I cannot set it to anything else.

To repro, I tried deleting the JSON setting file of the extension, and reinstall it (not a good repro in hindsight). The crashes went way then. So it is very likely that my existing JSON settings already had that invalid value.
